src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -2709,6 +2725,12 @@ public static DType Union(ref bool fError, DType leftType, DType rightType, bool
             leftType.AssertValid();
             rightType.AssertValid();
 
+            if ((leftType.IsSealed || rightType.IsSealed) && leftType != rightType)
```

Comment:
If leftType and rightType are not the same type, but are identical, should this work? For example, `[DropColumns(User, FullName), DropColumns(User, FullName)]`. We should have a test to validate that.

Reply:
I added tests for this, and tried various combinations.  On the surface, it seems like leftType != rightType shouldn't be enough, but somehow it is working.  Do we cache the DTypes and de-dup them?

Reply:
Oh, it looks like == has been overloaded for DType and parts.